processing/src/main/java/org/apache/druid/query/filter/ColumnComparisonDimFilter.java

```diff
@@ -114,10 +141,4 @@ public boolean equals(Object o)
         .collect(Collectors.toSet())
```

Comment:
Please fix `Sets.newHashSet(...collect(...))`. I've also opened #8241 about fixing this generically.

Reply:
I've changed the `DimFilter` interface to use `Set` instead of `HashSet` and adjusted this to no longer dupe the set, along with changing single dimension filters to use `ImmutableSet.of` instead of `Sets.newHashSet`